src/Common/Telemetry/Telemetry.php

```diff
@@ -462,13 +467,12 @@ public static function get_tec_telemetry_slugs() {
 	 * @return void
 	 */
 	public function register_tec_telemetry_plugins( $opted = NULL ) {
```

Comment:
We need the addition that I mentioned in my message in Slack: 

> register_tec_telemetry_plugins() is being called a bunch. It doesn't need to be. Let's make sure that is called only once.

Let's add a `Tribe__Cache::get()` and `Tribe__Cache::set()` (non-persistent) so we ensure this method is called only once. We could ignore the prevention if manually setting an opt-in.

Reply:
We additionally talked about ensuring that `::init()` in this method is only called during `admin_init`

Reply:
`boot()` runs on `tribe_plugins_loaded` and fires `tec_common_telemetry_preload`

`init()` runs on `admin_init`:
When `tec_common_telemetry_preload` fires -> `Provider::hook_telemetry_init()` is called -> it hooks `Provider::initialize_telemetry()` on `admin_init` -> then on `admin_init`, `initialize_telemetry` calls `Telemetry::init()`

Reply:
Ah, I didn't realize `init()` was being called on admin_init.